### DISCUSSION TITLE
Can't reach database Server

### DISCUSSION THREAD

**Discussion by devknight216** [prisma/prisma#21055] [Q&A]
### Question

I have a project that built with SvelteKit, Prisma, Supabase. It was deployed to Vercel.
Originally worked fine, but this morning, I have been going to execute it on my local and when I ran npm run dev command, it says "can't reach database server".
I erased prisma folder,  and executed "prisma init" to make schema file again and it worked.
And then I ran "prisma db pull" to get supabase database schema which I can see on Supabase project panel.
But I got the following error.
```
Prisma schema loaded from prisma\schema.prisma
Environment variables loaded from .env
Datasource "db": PostgreSQL database "postgres", schema "public" at "db.***arin.supabase.co:5432"

✖ Introspecting based on datasource defined in prisma\schema.prisma

Error: P1001

Can't reach database server at `db.***arin.supabase.co`:`5432` 

Please make sure your database server is running at `db.***arin.supabase.co`:`5432`.
```

**FYI, env file was set correctly so that I can connect DB with pgAdmin, there is no special characters in password string.**
Only Prisma has so many errors when I want to connect, fetch data, etc.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Windows10
- Database: Supabase Postgres
- Node.js version: 16.17.0

- Prisma Version: 5.2.0

**Comment by devknight216** [comment]
I temporarily solved this issue by adding "connect_timeout=300&pool_timeout=300" to the DB url.
But actually, DB only has one table which is not complex. **Why Prisma and Supabase need that kind of LONG TIMEOUT variable to connect?**
Which one has issue? Prisma or Supabase?

**Reply by janpio** [reply]
Most probably your internet connection between where you are, and Supabases hosts your database.
This should not take that long, and usually does not.
Are you close to the database, or is it on the other side of the globe?

**Reply by devknight216** [reply]
I am very close to the database region, EU-central-1. It is not the reason of the issue.

**Reply by devknight216** [reply]
I saw many discussions and issues that talks about the same issue. But it's not solved yet completely. Isn't there any other good way to solve this without increasing timeout?

**Reply by janpio** [reply]
If your connection to the database takes longer than the configured timeout, you will get an error about the timeout being reached.
There is nothing else that can be done - besides of course, making the connection faster.

**Reply by devknight216** [reply]
Even though I set the DB url with this "connect_timeout=300&pool_timeout=300", sometimes I faced "DB can't access" error message which makes my website has 500 error.

**Reply by MatteoGauthier** [reply]
Hey @devknight216 did you figure it out a better way to handle this issue ?

**Reply by Logannford** [reply]
+1, running into the same issue

**Reply by arheinjohnson** [reply]
+1

**Reply by ammanvedi** [reply]
seeing this also

**Reply by nkalmey-dev** [reply]
+1

**Reply by renanzucheratto** [reply]
+1

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
